plugins/pulp_rpm/plugins/migrations/0003_custom_metadata_on_scratchpad.py

```diff
@@ -14,6 +14,61 @@
 _log = logging.getLogger('pulp')
 
 
+def repository_working_dir(repo_id, mkdir=True):
+    """
+    Determines the repository's working directory. Individual plugin working
+    directories will be placed under this. If the mkdir argument is set to true,
+    the directory will be created as part of this call.
+    See the module-level docstrings for more information on the directory
+    structure.
+    @param mkdir: if true, this call will create the directory; otherwise the
+                  full path will just be generated
+    @type  mkdir: bool
+    @return: full path on disk
+    @rtype:  str
+    """
+    working_dir = os.path.join(_repo_working_dir(), repo_id)
+
+    if mkdir and not os.path.exists(working_dir):
+        os.makedirs(working_dir)
+
+    return working_dir
+
+
+def importer_working_dir(importer_type_id, repo_id, mkdir=True):
+    """
+    Determines the working directory for an importer to use for a repository.
+    If the mkdir argument is set to true, the directory will be created as
+    part of this call.
+    See the module-level docstrings for more information on the directory
+    structure.
+    @param mkdir: if true, this call will create the directory; otherwise the
+                  full path will just be generated
+    @type  mkdir: bool
+    @return: full path on disk to the directory the importer can use for the
+             given repository
+    @rtype:  str
+    """
+    repo_working_dir = repository_working_dir(repo_id, mkdir)
+    working_dir = os.path.join(repo_working_dir, 'importers', importer_type_id)
+
+    if mkdir and not os.path.exists(working_dir):
+        os.makedirs(working_dir)
+
+    return working_dir
+
+
+def _working_dir_root():
```

Comment:
I know these are small, but I'd make minimal doc blocks anyway. Our general rule is that if it's worth making something an independent function, it's worth taking 30 seconds to document its contract.